lib/package-managers/npm.js

```diff
@@ -107,25 +117,57 @@ module.exports = {
             npm.commands.listAsync(args || [], true); // silent:true
     },
 
-    latest: cint.partialAt(view, 1, 'dist-tags.latest'),
+    latest: function (packageName, currentVersion, includePrerelease) {
+        return view(packageName, 'dist-tags.latest')
+            .then(function (version) {
+                if (
+                    versionUtil.getPrecision(version) !== 'release' ||
+                    includePrerelease
+                ) {
+                    return version;
+                } else {
+                    return view(packageName, 'versions')
+                        .then(filterOutPrereleaseVersions)
+                        .then(_.last);
+                }
+            });
+    },
 
-    newest: function (packageName) {
+    newest: function (packageName, currentVersion, includePrerelease) {
         return view(packageName, 'time')
             .then(_.keys)
-            .then(_.last);
+            .then(function (versions) {
+                if (!includePrerelease) {
+                    versions = filterOutPrereleaseVersions(versions);
+                }
+                return _.last(versions);
+            });
     },
 
-    greatest: function (packageName) {
-        return view(packageName, 'versions').then(_.last);
+    greatest: function (packageName, currentVersion, includePrerelease) {
+        return view(packageName, 'versions')
+            .then(function (versions) {
+                if (!includePrerelease) {
+                    versions = filterOutPrereleaseVersions(versions);
+                }
+                return _.last(versions);
+            });
     },
 
-    greatestMajor: function (packageName, currentVersion) {
+    greatestMajor: function (packageName, currentVersion, includePrerelease) {
         return view(packageName, 'versions').then(function (versions) {
+            if (!includePrerelease) {
+                versions = filterOutPrereleaseVersions(versions);
+            }
             return versionUtil.findGreatestByLevel(versions, currentVersion, 'major');
```

Comment:
Since there is more logic here, you may want to not do it all on one line. Instead of overwriting an existing variable, however, declare a new variable such as `resultVersions`.